frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/useExecutionsForPipelineRun.ts

```diff
@@ -0,0 +1,19 @@
+import { useExecutionsFromContext } from '~/concepts/pipelines/apiHooks/mlmd/useExecutionsFromContext';
+import { usePipelineRunMlmdContext } from '~/concepts/pipelines/apiHooks/mlmd/usePipelineRunMlmdContext';
+import { isPipelineRunFinished } from '~/concepts/pipelines/apiHooks/usePipelineRunById';
+import { PipelineRunKFv2 } from '~/concepts/pipelines/kfTypes';
+import { Execution } from '~/third_party/mlmd';
+
+const useExecutionsForPipelineRun = (
+  run: PipelineRunKFv2 | null,
+): [executions: Execution[] | null, loaded: boolean, error?: Error] => {
+  const isFinished = isPipelineRunFinished(run);
+  // contextError means mlmd service is not available, no need to check executions
+  const [context, , contextError] = usePipelineRunMlmdContext(run?.run_id, !isFinished);
+  // executionsLoaded is the flag to show the spinner or not
+  const [executions, executionsLoaded] = useExecutionsFromContext(context, !isFinished);
```

Comment:
Looks like this is still called "FromContext" instead of "FromMlmdContext". Is this intentional? If so, why?

Reply:
Ah, I forgot to rename it, the hooks are mostly from the KubeFlow codebase, where they call everything context. Will rename it, thanks!